test/integration/deployment/deployment_test.go

```diff
@@ -498,3 +498,48 @@ func TestRollbackDeploymentRSNoRevision(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestFailedDeployment(t *testing.T) {
+	s, closeFn, rm, dc, informers, c := dcSetup(t)
+	defer closeFn()
+	name := "test-failed-deployment"
+	ns := framework.CreateTestingNamespace(name, s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+
+	deploymentName := "progress-check"
+	replicas := int32(1)
+	three := int32(3)
+	tester := &deploymentTester{t: t, c: c, deployment: newDeployment(deploymentName, ns.Name, replicas)}
+	tester.deployment.Spec.ProgressDeadlineSeconds = &three
+	var err error
+	tester.deployment, err = c.ExtensionsV1beta1().Deployments(ns.Name).Create(tester.deployment)
+	if err != nil {
+		t.Fatalf("failed to create deployment %q: %v", deploymentName, err)
+	}
+
+	// Start informer and controllers
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	informers.Start(stopCh)
+	go rm.Run(5, stopCh)
+	go dc.Run(5, stopCh)
+
+	if err = tester.waitForDeploymentUpdatedReplicasLTE(replicas); err != nil {
+		t.Fatalf("deployment %q failed to come up with %d replicas (with non-ready pods)", deploymentName, replicas)
```

Comment:
`err` should be printed too. However, `waitForDeploymentUpdatedReplicasLTE()` already returns similar message here, so returning raw error would be enough.